test/integration/dns_account_01_test.go

```diff
@@ -0,0 +1,383 @@
+		t.Fatalf("updating challenge: expected Incorrect TXT record error, got %s", prob.Detail)
+	}
+}
+
+func TestDNSAccount01MultipleTXTRecordsOneMatches(t *testing.T) {
+	t.Parallel()
+
+	domain := random_domain()
+	c, err := makeClient()
+	if err != nil {
+		t.Fatalf("creating client: %s", err)
+	}
+
+	idents := []acme.Identifier{{Type: "dns", Value: domain}}
+
+	order, err := c.Client.NewOrder(c.Account, idents)
+	if err != nil {
+		t.Fatalf("creating new order: %s", err)
+	}
+
+	authzURL := order.Authorizations[0]
+	auth, err := c.Client.FetchAuthorization(c.Account, authzURL)
+	if err != nil {
+		t.Fatalf("fetching authorization: %s", err)
+	}
+
+	chal, ok := auth.ChallengeMap[acme.ChallengeTypeDNSAccount01]
+	if !ok {
+		t.Skip("DNS-Account-01 is not offered, skipping test")
+	}
+
+	// Add multiple TXT records, one of which is correct
+	_, err = testSrvClient.AddDNSAccount01Response(c.Account.URL, domain, "wrong-digest-1")
+	if err != nil {
+		t.Fatalf("adding DNS response: %s", err)
+	}
+	_, err = testSrvClient.AddDNSAccount01Response(c.Account.URL, domain, chal.KeyAuthorization)
+	if err != nil {
+		t.Fatalf("adding DNS response: %s", err)
+	}
+	_, err = testSrvClient.AddDNSAccount01Response(c.Account.URL, domain, "wrong-digest-2")
+	if err != nil {
+		t.Fatalf("adding DNS response: %s", err)
+	}
+	t.Cleanup(func() {
+		_, _ = testSrvClient.RemoveDNSAccount01Response(c.Account.URL, domain)
+	})
+
+	chal, err = c.Client.UpdateChallenge(c.Account, chal)
+	if err != nil {
+		t.Fatalf("updating challenge: expected no error, got %s", err)
+	}
+}
+
+func TestDNSAccount01WildcardDomain(t *testing.T) {
+	t.Parallel()
+
+	hostDomain := randomDomain(t)
+	wildcardDomain := fmt.Sprintf("*.%s", randomDomain(t))
+
+	c, err := makeClient()
+	if err != nil {
+		t.Fatalf("creating client: %s", err)
+	}
+
+	idents := []acme.Identifier{
+		{Type: "dns", Value: hostDomain},
+		{Type: "dns", Value: wildcardDomain},
+	}
+
+	order, err := c.Client.NewOrder(c.Account, idents)
+	if err != nil {
+		t.Fatalf("creating new order: %s", err)
+	}
+
+	for _, authzURL := range order.Authorizations {
+		auth, err := c.Client.FetchAuthorization(c.Account, authzURL)
+		if err != nil {
+			t.Fatalf("fetching authorization: %s", err)
+		}
+
+		isWildcard := strings.HasPrefix(auth.Identifier.Value, "*.")
+		domain := auth.Identifier.Value
+		if isWildcard {
+			domain = strings.TrimPrefix(domain, "*.")
+		}
+
+		chal, ok := auth.ChallengeMap[acme.ChallengeTypeDNSAccount01]
+		if !ok {
+			t.Skip("DNS-Account-01 is not offered, skipping test")
+		}
+
+		_, err = testSrvClient.AddDNSAccount01Response(c.Account.URL, domain, chal.KeyAuthorization)
+		if err != nil {
+			t.Fatalf("adding DNS response: %s", err)
+		}
+		t.Cleanup(func() {
+			_, _ = testSrvClient.RemoveDNSAccount01Response(c.Account.URL, domain)
+		})
+
+		chal, err = c.Client.UpdateChallenge(c.Account, chal)
+		if err != nil {
+			t.Fatalf("updating challenge: %s", err)
+		}
+	}
+
```

Comment:
Same comment here, I think this test would be better served by simply checking the status of the Order or Authorization, rather than going through finalization.